src/modules/python/pbs/v1/_svr_types.py

```diff
@@ -1365,3 +1365,136 @@ def __next__(self):
+        super().__setattr__(name, value)
+    #: m(__setattr__)
+
+    def extract_flags_str(self):
+        """returns the string values from the attribute flags."""
+        lst = []
+        for mask, value in _pbs_v1.REVERSE_ATR_VFLAGS.items():
+            if self.flags & mask:
+                lst.append(value)
+        return lst
+    #: m(extract_flags_str)
+
+    def extract_flags_int(self):
+        """returns the integer values from the attribute flags."""
+        lst = []
+        for mask, value in _pbs_v1.REVERSE_ATR_VFLAGS.items():
+            if self.flags & mask:
+                lst.append(mask)
+        return lst
+    #: m(extract_flags_int)
+
+    def tup(self):
+        return self.name, self.resource, self.value, self.op, self.flags, self.sisters
+    #: m(tup)
+
+_server_attribute._connect_server = PbsAttributeDescriptor(
+    _server_attribute, '_connect_server', "", (str,))
+#: C(_server_attribute)
+
+# This exposes pbs.server_attribute() to be callable in a hook script
+server_attribute = _server_attribute
+
+#:------------------------------------------------------------------------
+#                  MANAGEMENT TYPE
+#:-------------------------------------------------------------------------
+class _management:
+    """
+    This represents a management operation.
+    """
+    attributes = PbsReadOnlyDescriptor('attributes', {})
+    _attributes_hook_set = {}
+
+    def __init__(self, cmd, objtype, objname, request_time, reply_code,
+        reply_auxcode, reply_choice, reply_text,
+        attribs, connect_server=None):
+        """__init__"""
+        self.cmd = cmd
+        self.objtype = objtype
+        self.objname = objname
+        self.request_time = request_time
+        self.reply_code = reply_code
+        self.reply_auxcode = reply_auxcode
+        self.reply_choice = reply_choice
+        self.reply_text = reply_text
+        self.attribs = attribs
+        self._readonly = True
+        self._connect_server = connect_server
+    #: m(__init__)
+
+    def __str__(self):
+        """String representation of the object"""
+        return "%s:%s:%s" % (
+            _pbs_v1.REVERSE_MGR_CMDS.get(self.cmd, self.cmd),
+            _pbs_v1.REVERSE_MGR_OBJS.get(self.objtype, self.objtype),
+            self.objname
+            )
+    #: m(__str__)
+
+    def __setattr__(self, name, value):
+        if _pbs_v1.in_python_mode():
+            raise BadAttributeValueError(
+                "'%s' attribute in the management object is readonly" % (name,))
+        super().__setattr__(name, value)
+    #: m(__setattr__)
+
+_management.cmd = PbsAttributeDescriptor(_management, 'cmd', None, (int,))
+_management.objtype = PbsAttributeDescriptor(_management, 'objtype', None, (int,))
+_management.objname = PbsAttributeDescriptor(_management, 'objname', "", (str,))
+_management._connect_server = PbsAttributeDescriptor(
+    _management, '_connect_server', "", (str,))
```

Comment:
Do we need these lines?  What purpose do they server?

Reply:
The response provided for the `_server_attribute` class answers the reason to keep the `PbsAttributeDescriptor` for `_connect_server`, and that seems reasonable to me.  I do still wonder about the other attributes in the `_management` class.  We only added those calls because the existing classes we used as examples has them and we did not understand their purpose.  However, we did not add them for all `_management` class attributes, so we should either expand the calls to include all attributes or remove all but the `_connect_server` attribute.  Again, not understanding the purpose of `PbsAttributeDescriptor`, I'm not sure which is appropriate.

Reply:
The PbsAttributeDescriptor is a way to individually control an attribute per instance, especially if some attributes are to behave a certain way, like making them read-only.